clean.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+
+usage()
+{
+    echo "Usage: clean [options]"
+    echo "Cleans the local dev environment."
+    echo
+    echo "  -b     Delete bin directory"
+    echo "  -p     Delete packages directory and NuGet package caches"
+    echo "  -a     Remove all untracked files in the working tree and delete NuGet package caches"
+    exit 1
+}
+
+# Parse arguments
+
+if [ $# == 0 ]
+then
+    echo "Please specify at least one option."
+    echo "Use 'clean -h' for help."
+    exit 1
+fi
+
+while [[ $# > 0 ]]
+do
+    opt="$1"
+    case $opt in
+        -h|--help)
+        usage
+        ;;
+        -b)
+        CleanBinDir=true
+        ;;
+        -p)
+        CleanPackagesDir=true
+        CleanPackagesCache=true
+        ;;
+        -a)
+        CleanWorkingTree=true
+        CleanPackagesCache=true
+        ;;
+        *)
+        echo "Unrecognized argument '$opt'"
+        echo "Use 'clean -h' for help."
+        exit 1
+        ;;
+    esac
+    shift
+done
+
+WorkingTreeRoot="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+if [ "$CleanBinDir" == true ] && [ "$CleanWorkingTree" != true ]
+then
+    echo "Removing bin directory"
+    rm -rf $WorkingTreeRoot/bin
+fi
+
+if [ "$CleanPackagesDir" == true ] && [ "$CleanWorkingTree" != true ]
+then
+    echo "Removing packages directory"
+    rm -rf $WorkingTreeRoot/packages
```

Comment:
My main concern with using the Clean MSBuild target is that it has a single set behavior, whereas one of the main benefits of this script is that it allows multiple options (e.g. just removing the bin directory, just removing packages, or cleaning everything).  I don't know if I like the idea of coupling one of the options of clean.sh to the Clean target of build.proj.  Currently, the Clean target just removes the bin directory, but it's possible that at some point we'll modify it to also remove packages (for example), in which case we'd also need to update clean.sh.  So, from a code maintenance perspective, I think it might hurt more than it would help.

It's also just a single line that's duplicated, so I don't see it as a concern from a complexity perspective.

And Eric, I think we want to have a separate clean.sh script for convenience and consistency with the existing workflow that we have in TFS.  Even if we do want to utilize MSBuild to avoid the duplication of logic, I think it would be better to use Wes's suggestion of having clean.sh invoke MSBuild.